tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dsl/DSLResultTest.java

```diff
@@ -0,0 +1,104 @@
+import org.dataflowanalysis.analysis.dsl.variable.ConstraintVariable;
+import org.dataflowanalysis.analysis.dsl.constraint.ConstraintDSL;
+import org.dataflowanalysis.analysis.dsl.selectors.Intersection;
+import org.dataflowanalysis.analysis.dsl.selectors.CharacteristicsSelectorData;
+import org.dataflowanalysis.analysis.dsl.selectors.DataCharacteristicsSelector;
+import org.dataflowanalysis.analysis.dsl.selectors.VertexCharacteristicsSelector;
+import org.dataflowanalysis.analysis.dsl.variable.ConstraintVariableReference;
+import org.dataflowanalysis.analysis.pcm.core.user.UserPCMVertex;
+import org.dataflowanalysis.analysis.pcm.dsl.PCMVertexType;
+import org.dataflowanalysis.analysis.tests.BaseTest;
+import org.dataflowanalysis.analysis.tests.constraint.data.ConstraintData;
+import org.dataflowanalysis.analysis.tests.constraint.data.ConstraintViolations;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+public class DSLResultTest extends BaseTest {
+    private static final Logger logger = Logger.getLogger(DSLResultTest.class);
+
+    @Test
+    public void testDSL() {
+        var constraint = new ConstraintDSL()
+                .ofData()
+                .withLabel("DataSensitivity", List.of("Personal"))
+                .ofNode()
+                .neverFlows()
+                .toVertex()
+                .withCharacteristic("ServerLocation", "nonEU")
+                .create();
+
+        evaluateAnalysis(constraint, internationalOnlineShopAnalysis, ConstraintViolations.internationalOnlineShopViolations);
+    }
+
+    @Test
+    public void testVariableDSL() {
+        AnalysisConstraint constraint = new ConstraintDSL()
+                .ofData()
+                .withLabel("GrantedRoles", ConstraintVariable.of("grantedRoles"))
+                .neverFlows()
+                .toVertex()
+                .withCharacteristic("AssignedRoles", ConstraintVariable.of("assignedRoles"))
+                .where()
+                .isNotEmpty(ConstraintVariable.of("grantedRoles"))
+                .isNotEmpty(ConstraintVariable.of("assignedRoles"))
+                .isEmpty(Intersection.of(ConstraintVariable.of("grantedRoles"), ConstraintVariable.of("assignedRoles")))
+                .create();
+
+        evaluateAnalysis(constraint, travelPlannerAnalysis, ConstraintViolations.travelPlannerViolations);
+    }
+
+    @Test
+    public void testQueryDSL() {
+        AnalysisQuery query = new QueryDSL()
+                .ofNode()
+                .withType(PCMVertexType.USER)
+                .build();
+        FlowGraphCollection flowGraphCollection = this.travelPlannerAnalysis.findFlowGraphs();
+        List<DSLResult> results = query.query(flowGraphCollection);
+        List<? extends AbstractVertex<?>> queriedVertices = results.stream()
+                .map(DSLResult::getViolatingVertices)
```

Comment:
I suppose that the name `getViolatingVertices` originates from constraint violations? At least for query results it is somewhat misleading. Maybe a simple `getVertices` would already solve this?

Reply:
Fixed in 24757fe1f865